src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/NotificationShimFactory.cs

```diff
@@ -42,6 +42,7 @@ private void Initialize(SafeWaitHandle hEvent)
         // TODO: Instantiate all the locks (critical sections)
     }
 
+    [System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessage("Trimming", "IL2050", Justification = "Leave me alone")]
```

Comment:
As for the code itself - if the PInvoke is indeed OK (meaning no COM will be triggered at runtime - I don't know), the we typically do one other thing:
* Suppressing at the method level is very coarse - unfortunately it's the only currently available way (since attributes can't be applied at IL level), and thus suppressing on large methods is problematic for future code changes (the suppression could hide a problem introduced later on)
* For this we typically create a local function which does the one "dangerous" thing and suppress on it.
* It is sort of "tooling's inability forcing the code to be structured in a specific way" which is unfortunate, but so far it wasn't that bad.